--- ./feeds/luci/applications/luci-openvpn/root/etc/config/openvpn_recipes	2014-08-24 12:43:12.000000000 +0800
+++ ./feeds/luci/applications/luci-openvpn/root/etc/config/openvpn_recipes	2014-07-09 20:11:29.000000000 +0800
@@ -8,10 +8,9 @@ config openvpn_recipe server_tun_ptp
 	option ifconfig			"10.0.0.1 10.0.0.2"
 	option secret			"shared-secret.key"
 	option keepalive		"10 60"
-	option comp_lzo			"1"
+	option comp_lzo			"yes"
 	option verb			"3"
-        option mssfix			"1420"
-	option management		"127.0.0.1 31194"
+	option mssfix			"1420"
 
 #
 # Routed point-to-point client
@@ -24,9 +23,8 @@ config openvpn_recipe client_tun_ptp
 	option ifconfig			"10.0.0.2 10.0.0.1"
 	option secret			"shared-secret.key"
 	option nobind			"1"
-	option comp_lzo			"1"
+	option comp_lzo			"yes"
 	option verb			"3"
-	option management		"127.0.0.1 31194"
 
 #
 # Routed multi-client server
@@ -41,10 +39,9 @@ config openvpn_recipe server_tun
 	option key 			"server.key"
 	option dh 			"dh1024.pem"
 	option keepalive		"10 60"
-	option comp_lzo			"1"
+	option comp_lzo			"yes"
 	option verb			"3"
-        option mssfix			"1420"
-	option management		"127.0.0.1 31194"
+	option mssfix			"1420"
 
 #
 # Routed client
@@ -57,14 +54,13 @@ config openvpn_recipe client_tun
 	list   remote			"vpnserver.example.org"
 	option pkcs12			"my_client.p12"
 	option remote_cert_tls		"server"
-	option comp_lzo			"1"
+	option comp_lzo			"yes"
 	option nobind			"1"
 	option persist_key		"1"
 	option persist_tun		"1"
 	option verb			"3"
 	option reneg_sec		"0"
 	option float			"1"
-	option management		"127.0.0.1 31194"
 
 #
 # Multi-client ethernet bridge server
@@ -79,10 +75,9 @@ config openvpn_recipe server_tap_bridge
 	option key 			"server.key"
 	option dh 			"dh1024.pem"
 	option keepalive		"10 60"
-	option comp_lzo			"1"
+	option comp_lzo			"yes"
 	option verb			"3"
-        option mssfix			"1420"
-	option management		"127.0.0.1 31194"
+	option mssfix			"1420"
 
 #
 # Ethernet bridge client
@@ -98,10 +93,9 @@ config openvpn_recipe client_tap_bridge
 	option key 			"my_client.key"
 	option dh 			"dh1024.pem"
 	option remote_cert_tls		"server"
-	option comp_lzo			"1"
+	option comp_lzo			"yes"
 	option nobind			"1"
 	option persist_key		"1"
 	option verb			"3"
 	option reneg_sec		"0"
 	option float			"1"
-	option management		"127.0.0.1 31194"
