--- ./feeds/luci/contrib/package/luci-addons/Makefile	2014-08-24 12:43:12.000000000 +0800
+++ ./feeds/luci/contrib/package/luci-addons/Makefile	2014-08-24 15:02:20.545315511 +0800
@@ -181,7 +181,7 @@ $(eval $(call application,ushare,uShare
 $(eval $(call application,hd-idle,Hard Disk Idle Spin-Down module,hd-idle))
 $(eval $(call application,tinyproxy,Tinyproxy - HTTP(S)-Proxy configuration,tinyproxy))
 $(eval $(call application,polipo,LuCI Support for the Polipo Proxy,polipo))
-$(eval $(call application,openvpn,LuCI Support for OpenVPN,openvpn @BROKEN))
+$(eval $(call application,openvpn,LuCI Support for OpenVPN,openvpn))
 $(eval $(call application,p2pblock,LuCI Support for the Freifunk P2P-Block addon,luci-app-firewall freifunk-p2pblock))
 $(eval $(call application,multiwan,LuCI Support for the OpenWrt MultiWAN agent,luci-app-firewall multiwan))
 $(eval $(call application,wol,LuCI Support for Wake-on-LAN,etherwake))
