--- ./target/linux/ramips/Makefile	2014-08-24 12:42:14.000000000 +0800
+++ ./target/linux/ramips/Makefile	2014-08-27 20:43:01.241391584 +0800
@@ -11,12 +11,12 @@ BOARD:=ramips
 BOARDNAME:=Ralink RT288x/RT3xxx
 SUBTARGETS:=rt305x mt7620a mt7620n rt3883 rt288x
 FEATURES:=squashfs gpio
-
+CPU_TYPE=24kec
 KERNEL_PATCHVER:=3.10
 
 include $(INCLUDE_DIR)/target.mk
 DEFAULT_PACKAGES += \
-	kmod-leds-gpio kmod-gpio-button-hotplug wpad-mini swconfig
+	kmod-leds-gpio kmod-gpio-button-hotplug wpad swconfig
 
 define Target/Description
 	Build firmware images for Ralink RT288x/RT3xxx based boards.
