--- ./config/Config-kernel.in	2014-08-31 13:47:57.000000000 +0800
+++ ./config/Config-kernel.in	2014-08-31 13:55:27.482987845 +0800
@@ -11,7 +11,7 @@ config KERNEL_PRINTK
 config KERNEL_CRASHLOG
 	bool "Crash logging"
 	depends on !(arm || powerpc || sparc || TARGET_uml)
-	default y
+	default n
 
 config KERNEL_SWAP
 	bool "Support for paging of anonymous memory (swap)"
@@ -19,7 +19,7 @@ config KERNEL_SWAP
 
 config KERNEL_DEBUG_FS
 	bool "Compile the kernel with Debug FileSystem enabled"
-	default y
+	default n
 	help
 	  debugfs is a virtual file system that kernel developers use to put
 	  debugging files into. Enable this option to be able to read and
@@ -39,7 +39,7 @@ config KERNEL_PROFILING
 
 config KERNEL_KALLSYMS
 	bool "Compile the kernel with symbol table information"
-	default y
+	default n
 	help
 	  This will give you more information in stack traces from kernel oopses
 
@@ -63,7 +63,7 @@ config KERNEL_DEBUG_KERNEL
 
 config KERNEL_DEBUG_INFO
 	bool "Compile the kernel with debug information"
-	default y
+	default n
 	select KERNEL_DEBUG_KERNEL
 	help
 	  This will compile your kernel and modules with debug information.
@@ -115,7 +115,7 @@ config KERNEL_DIRECT_IO
 
 config KERNEL_MAGIC_SYSRQ
 	bool "Compile the kernel with SysRq support"
-	default y
+	default n
 
 config KERNEL_COREDUMP
 	bool
@@ -123,7 +123,7 @@ config KERNEL_COREDUMP
 config KERNEL_ELF_CORE
 	bool "Enable process core dump support"
 	select KERNEL_COREDUMP
-	default y
+	default n
 
 config KERNEL_PROVE_LOCKING
 	bool "Enable kernel lock checking"
@@ -144,9 +144,11 @@ config KERNEL_SLABINFO
 	select KERNEL_SLUB_DEBUG
 	select KERNEL_SLUB_DEBUG_ON
 	bool "Enable /proc slab debug info"
+	default n
 
 config KERNEL_PROC_PAGE_MONITOR
 	bool "Enable /proc page monitoring"
+	default n
 
 config KERNEL_RELAY
 	bool
