--- ./toolchain/gcc/common.mk	2014-09-06 20:31:20.000000000 +0800
+++ ./toolchain/gcc/common.mk	2014-09-14 23:49:44.777543516 +0800
@@ -34,13 +34,13 @@ ifeq ($(findstring linaro, $(CONFIG_GCC_
       PKG_COMP:=bz2
     endif
     ifeq ($(CONFIG_GCC_VERSION),"4.8-linaro")
-      PKG_REV:=4.8-2014.04
-      PKG_VERSION:=4.8.3
+      PKG_REV:=4.8-2014.08
+      PKG_VERSION:=4.8.4
       PKG_VERSION_MAJOR:=4.8
-      PKG_MD5SUM:=5ba2f3a449b1658ccc09d27cc7ab3c03
+      PKG_MD5SUM:=6f506cd7000b07f69992c18902170955
       PKG_COMP:=xz
     endif
-    PKG_SOURCE_URL:=http://launchpad.net/gcc-linaro/$(PKG_VERSION_MAJOR)/$(PKG_REV)/+download/
+    PKG_SOURCE_URL:=http://releases.linaro.org/14.08/components/toolchain/gcc-linaro/$(PKG_VERSION_MAJOR)/
     PKG_SOURCE:=$(PKG_NAME)-linaro-$(PKG_REV).tar.$(PKG_COMP)
     GCC_DIR:=gcc-linaro-$(PKG_REV)
     HOST_BUILD_DIR:=$(BUILD_DIR_TOOLCHAIN)/$(GCC_DIR)
