--- ./package/kernel/gpio-button-hotplug/Makefile	2014-11-22 16:22:34.000000000 +0800
+++ ./package/kernel/gpio-button-hotplug/Makefile	2014-11-22 16:22:46.597899240 +0800
@@ -18,7 +18,7 @@ define KernelPackage/gpio-button-hotplug
   DEPENDS:=@!LINUX_3_3
   TITLE:=Simple GPIO Button Hotplug driver
   FILES:=$(PKG_BUILD_DIR)/gpio-button-hotplug.ko
-  AUTOLOAD:=$(call AutoLoad,30,gpio-button-hotplug,1)
+  AUTOLOAD:=$(call AutoProbe,gpio-button-hotplug)
   KCONFIG:=
 endef

--- ./package/kernel/linux/modules/crypto.mk	2014-11-22 16:16:46.000000000 +0800
+++ ./package/kernel/linux/modules/crypto.mk	2014-11-22 16:30:47.097004788 +0800
@@ -43,7 +43,7 @@ define KernelPackage/crypto-hash
   TITLE:=CryptoAPI hash support
   KCONFIG:=CONFIG_CRYPTO_HASH
   FILES:=$(LINUX_DIR)/crypto/crypto_hash.ko
-  AUTOLOAD:=$(call AutoLoad,02,crypto_hash,1)
+  AUTOLOAD:=$(call AutoProbe,crypto_hash)
   $(call AddDepends/crypto)
 endef
 
@@ -255,7 +255,7 @@ define KernelPackage/crypto-arc4
   TITLE:=ARC4 (RC4) cipher CryptoAPI module
   KCONFIG:=CONFIG_CRYPTO_ARC4
   FILES:=$(LINUX_DIR)/crypto/arc4.ko
-  AUTOLOAD:=$(call AutoLoad,09,arc4)
+  AUTOLOAD:=$(call AutoProbe,arc4)
   $(call AddDepends/crypto)
 endef
 
@@ -345,7 +345,7 @@ define KernelPackage/crypto-ecb
   DEPENDS:=+kmod-crypto-manager
   KCONFIG:=CONFIG_CRYPTO_ECB
   FILES:=$(LINUX_DIR)/crypto/ecb.ko
-  AUTOLOAD:=$(call AutoLoad,09,ecb)
+  AUTOLOAD:=$(call AutoProbe,ecb)
   $(call AddDepends/crypto)
 endef
 
@@ -406,7 +406,7 @@ define KernelPackage/crypto-sha1
   DEPENDS:=+kmod-crypto-hash
   KCONFIG:=CONFIG_CRYPTO_SHA1
   FILES:=$(LINUX_DIR)/crypto/sha1_generic.ko
-  AUTOLOAD:=$(call AutoLoad,09,sha1_generic)
+  AUTOLOAD:=$(call AutoProbe,sha1_generic)
   $(call AddDepends/crypto)
 endef
 
--- ./package/kernel/linux/modules/usb.mk	2014-11-22 16:24:15.000000000 +0800
+++ ./package/kernel/linux/modules/usb.mk	2014-11-22 17:12:09.072645554 +0800
@@ -19,7 +19,7 @@ define KernelPackage/usb-core
   FILES:= \
 	$(LINUX_DIR)/drivers/usb/core/usbcore.ko \
 	$(LINUX_DIR)/drivers/usb/usb-common.ko
-  AUTOLOAD:=$(call AutoLoad,20,usb-common usbcore,1)
+  AUTOLOAD:=$(call AutoProbe,usb-common usbcore)
   $(call AddDepends/nls)
 endef
 
@@ -331,7 +331,7 @@ define KernelPackage/usb-ohci
 	CONFIG_USB_OHCI_HCD_OMAP3=y \
 	CONFIG_USB_OHCI_HCD_PLATFORM=y
   FILES:=$(LINUX_DIR)/drivers/usb/host/ohci-hcd.ko
-  AUTOLOAD:=$(call AutoLoad,50,ohci-hcd,1)
+  AUTOLOAD:=$(call AutoProbe,ohci-hcd)
   $(call AddDepends/usb)
 endef
 
@@ -397,10 +397,10 @@ ifeq ($(strip $(call CompareKernelPatchV
   ifneq ($(wildcard $(LINUX_DIR)/drivers/usb/host/ehci-orion.ko),)
     FILES+=$(LINUX_DIR)/drivers/usb/host/ehci-orion.ko
   endif
-  AUTOLOAD:=$(call AutoLoad,40,ehci-hcd ehci-platform ehci-orion,1)
+  AUTOLOAD:=$(call AutoProbe,ehci-hcd ehci-platform ehci-orion)
 else
   FILES:=$(LINUX_DIR)/drivers/usb/host/ehci-hcd.ko
-  AUTOLOAD:=$(call AutoLoad,40,ehci-hcd,1)
+  AUTOLOAD:=$(call AutoProbe,ehci-hcd)
 endif
   $(call AddDepends/usb)
 endef
@@ -445,7 +445,7 @@ else
 	$(LINUX_DIR)/drivers/staging/dwc2/dwc2.ko \
 	$(LINUX_DIR)/drivers/staging/dwc2/dwc2_platform.ko
 endif
-  AUTOLOAD:=$(call AutoLoad,54,dwc2 dwc2_platform,1)
+  AUTOLOAD:=$(call AutoProbe,dwc2 dwc2_platform)
   $(call AddDepends/usb)
 endef
 
@@ -1430,7 +1430,7 @@ define KernelPackage/usb3
 	CONFIG_USB_XHCI_HCD_DEBUGGING=n
   FILES:= \
 	$(LINUX_DIR)/drivers/usb/host/xhci-hcd.ko
-  AUTOLOAD:=$(call AutoLoad,54,xhci-hcd,1)
+  AUTOLOAD:=$(call AutoProbe,xhci-hcd)
   $(call AddDepends/usb)
 endef

--- ./package/kernel/linux/modules/other.mk	2014-11-22 19:01:11.000000000 +0800
+++ ./package/kernel/linux/modules/other.mk	2014-11-22 19:01:23.731020732 +0800
@@ -91,7 +91,7 @@ define KernelPackage/eeprom-93cx6
   TITLE:=EEPROM 93CX6 support
   KCONFIG:=CONFIG_EEPROM_93CX6
   FILES:=$(LINUX_DIR)/drivers/misc/eeprom/eeprom_93cx6.ko
-  AUTOLOAD:=$(call AutoLoad,20,eeprom_93cx6)
+  AUTOLOAD:=$(call AutoProbe,eeprom_93cx6)
 endef
 
 define KernelPackage/eeprom-93cx6/description
@@ -732,7 +732,7 @@ else
 	$(LINUX_DIR)/drivers/staging/zsmalloc/zsmalloc.ko \
 	$(LINUX_DIR)/drivers/staging/zram/zram.ko
 endif
-  AUTOLOAD:=$(call AutoLoad,20,zsmalloc zram)
+  AUTOLOAD:=$(call AutoProbe,zsmalloc zram)
 endef
 
 define KernelPackage/zram/description

--- ./package/kernel/linux/modules/leds.mk	2014-11-22 17:42:59.000000000 +0800
+++ ./package/kernel/linux/modules/leds.mk	2014-11-22 19:10:35.236601162 +0800
@@ -13,7 +13,7 @@ define KernelPackage/leds-gpio
   DEPENDS:= @GPIO_SUPPORT
   KCONFIG:=CONFIG_LEDS_GPIO
   FILES:=$(LINUX_DIR)/drivers/leds/leds-gpio.ko
-  AUTOLOAD:=$(call AutoLoad,60,leds-gpio,1)
+  AUTOLOAD:=$(call AutoProbe,leds-gpio)
 endef
 
 define KernelPackage/leds-gpio/description
@@ -28,7 +28,7 @@ define KernelPackage/ledtrig-heartbeat
   TITLE:=LED Heartbeat Trigger
   KCONFIG:=CONFIG_LEDS_TRIGGER_HEARTBEAT
   FILES:=$(LINUX_DIR)/drivers/leds/$(if $(call kernel_patchver_ge,3.10),trigger/)ledtrig-heartbeat.ko
-  AUTOLOAD:=$(call AutoLoad,50,ledtrig-heartbeat)
+  AUTOLOAD:=$(call AutoProbe,ledtrig-heartbeat)
 endef
 
 define KernelPackage/ledtrig-gpio/description
@@ -43,7 +43,7 @@ define KernelPackage/ledtrig-gpio
   TITLE:=LED GPIO Trigger
   KCONFIG:=CONFIG_LEDS_TRIGGER_GPIO
   FILES:=$(LINUX_DIR)/drivers/leds/$(if $(call kernel_patchver_ge,3.10),trigger/)ledtrig-gpio.ko
-  AUTOLOAD:=$(call AutoLoad,50,ledtrig-gpio)
+  AUTOLOAD:=$(call AutoProbe,ledtrig-gpio)
 endef
 
 define KernelPackage/ledtrig-gpio/description
@@ -58,7 +58,7 @@ define KernelPackage/ledtrig-morse
   TITLE:=LED Morse Trigger
   KCONFIG:=CONFIG_LEDS_TRIGGER_MORSE
   FILES:=$(LINUX_DIR)/drivers/leds/ledtrig-morse.ko
-  AUTOLOAD:=$(call AutoLoad,50,ledtrig-morse)
+  AUTOLOAD:=$(call AutoProbe,ledtrig-morse)
 endef
 
 define KernelPackage/ledtrig-morse/description
@@ -73,7 +73,7 @@ define KernelPackage/ledtrig-netdev
   TITLE:=LED NETDEV Trigger
   KCONFIG:=CONFIG_LEDS_TRIGGER_NETDEV
   FILES:=$(LINUX_DIR)/drivers/leds/ledtrig-netdev.ko
-  AUTOLOAD:=$(call AutoLoad,50,ledtrig-netdev)
+  AUTOLOAD:=$(call AutoProbe,ledtrig-netdev)
 endef
 
 define KernelPackage/ledtrig-netdev/description
@@ -89,7 +89,7 @@ define KernelPackage/ledtrig-netfilter
   DEPENDS:=kmod-ipt-core
   KCONFIG:=CONFIG_NETFILTER_XT_TARGET_LED
   FILES:=$(LINUX_DIR)/net/netfilter/xt_LED.ko
-  AUTOLOAD:=$(call AutoLoad,50,xt_LED)
+  AUTOLOAD:=$(call AutoProbe,xt_LED)
 endef
 
 define KernelPackage/ledtrig-netfilter/description
@@ -110,7 +110,7 @@ define KernelPackage/ledtrig-usbdev
   DEPENDS:=@USB_SUPPORT kmod-usb-core
   KCONFIG:=CONFIG_LEDS_TRIGGER_USBDEV
   FILES:=$(LINUX_DIR)/drivers/leds/ledtrig-usbdev.ko
-  AUTOLOAD:=$(call AutoLoad,50,ledtrig-usbdev)
+  AUTOLOAD:=$(call AutoProbe,ledtrig-usbdev)
 endef
 
 define KernelPackage/ledtrig-usbdev/description
@@ -125,7 +125,7 @@ define KernelPackage/ledtrig-default-on
   TITLE:=LED Default ON Trigger
   KCONFIG:=CONFIG_LEDS_TRIGGER_DEFAULT_ON
   FILES:=$(LINUX_DIR)/drivers/leds/$(if $(call kernel_patchver_ge,3.10),trigger/)ledtrig-default-on.ko
-  AUTOLOAD:=$(call AutoLoad,50,ledtrig-default-on,1)
+  AUTOLOAD:=$(call AutoProbe,ledtrig-default-on)
 endef
 
 define KernelPackage/ledtrig-default-on/description
@@ -140,7 +140,7 @@ define KernelPackage/ledtrig-timer
   TITLE:=LED Timer Trigger
   KCONFIG:=CONFIG_LEDS_TRIGGER_TIMER
   FILES:=$(LINUX_DIR)/drivers/leds/$(if $(call kernel_patchver_ge,3.10),trigger/)ledtrig-timer.ko
-  AUTOLOAD:=$(call AutoLoad,50,ledtrig-timer,1)
+  AUTOLOAD:=$(call AutoProbe,ledtrig-timer)
 endef
 
 define KernelPackage/ledtrig-timer/description
@@ -157,7 +157,7 @@ define KernelPackage/ledtrig-oneshot
   DEPENDS:=@!LINUX_3_3
   KCONFIG:=CONFIG_LEDS_TRIGGER_ONESHOT
   FILES:=$(LINUX_DIR)/drivers/leds/$(if $(call kernel_patchver_ge,3.10),trigger/)ledtrig-oneshot.ko
-  AUTOLOAD:=$(call AutoLoad,50,ledtrig-oneshot)
+  AUTOLOAD:=$(call AutoProbe,ledtrig-oneshot)
 endef
 
 define KernelPackage/ledtrig-oneshot/description
@@ -174,7 +174,7 @@ define KernelPackage/leds-tlc59116
   DEPENDS:=@TARGET_mvebu kmod-i2c-core
   KCONFIG:=CONFIG_LEDS_TLC59116
   FILES:=$(LINUX_DIR)/drivers/leds/leds-tlc59116.ko
-  AUTOLOAD:=$(call AutoLoad,60,leds-tlc59116,1)
+  AUTOLOAD:=$(call AutoProbe,leds-tlc59116)
 endef
 
 define KernelPackage/leds-tlc59116/description

--- ./package/kernel/linux/modules/netdevices.mk	2014-11-22 23:40:52.000000000 +0800
+++ ./package/kernel/linux/modules/netdevices.mk	2014-11-23 01:06:49.646157741 +0800
@@ -685,7 +685,7 @@ define KernelPackage/ifb
 	CONFIG_IFB \
 	CONFIG_NET_CLS=y
   FILES:=$(LINUX_DIR)/drivers/net/ifb.ko
-  AUTOLOAD:=$(call AutoLoad,34,ifb)
+  AUTOLOAD:=$(call AutoProbe,ifb)
 endef
 
 define KernelPackage/ifb/description

--- ./package/kernel/linux/modules/netsupport.mk	2014-11-23 01:43:17.000000000 +0800
+++ ./package/kernel/linux/modules/netsupport.mk	2014-11-23 02:20:50.874730403 +0800
@@ -352,7 +352,7 @@ define KernelPackage/iptunnel
   KCONFIG:= \
 	CONFIG_NET_IP_TUNNEL
   FILES:=$(LINUX_DIR)/net/ipv4/ip_tunnel.ko
-  AUTOLOAD:=$(call AutoLoad,31,ip_tunnel)
+  AUTOLOAD:=$(call AutoProbe,ip_tunnel)
 endef
 
 define KernelPackage/iptunnel/description
@@ -369,7 +369,7 @@ define KernelPackage/iptunnel4
   KCONFIG:= \
 	CONFIG_INET_TUNNEL
   FILES:=$(LINUX_DIR)/net/ipv4/tunnel4.ko
-  AUTOLOAD:=$(call AutoLoad,31,tunnel4)
+  AUTOLOAD:=$(call AutoProbe,tunnel4)
 endef
 
 define KernelPackage/iptunnel4/description
@@ -386,7 +386,7 @@ define KernelPackage/iptunnel6
   KCONFIG:= \
 	CONFIG_INET6_TUNNEL
   FILES:=$(LINUX_DIR)/net/ipv6/tunnel6.ko
-  AUTOLOAD:=$(call AutoLoad,31,tunnel6)
+  AUTOLOAD:=$(call AutoProbe,tunnel6)
 endef
 
 define KernelPackage/iptunnel6/description
@@ -407,7 +407,7 @@ define KernelPackage/ipv6
 	CONFIG_IPV6_PIMSM_V2=n \
 	CONFIG_IPV6_SUBTREES=y
   FILES:=$(LINUX_DIR)/net/ipv6/ipv6.ko
-  AUTOLOAD:=$(call AutoLoad,20,ipv6)
+  AUTOLOAD:=$(call AutoProbe,ipv6)
 endef
 
 define KernelPackage/ipv6/description
@@ -424,7 +424,7 @@ define KernelPackage/sit
   KCONFIG:=CONFIG_IPV6_SIT \
 	CONFIG_IPV6_SIT_6RD=y
   FILES:=$(LINUX_DIR)/net/ipv6/sit.ko
-  AUTOLOAD:=$(call AutoLoad,32,sit)
+  AUTOLOAD:=$(call AutoProbe,sit)
 endef
 
 define KernelPackage/sit/description
@@ -440,7 +440,7 @@ define KernelPackage/ip6-tunnel
   DEPENDS:= +kmod-ipv6 +kmod-iptunnel6
   KCONFIG:= CONFIG_IPV6_TUNNEL
   FILES:=$(LINUX_DIR)/net/ipv6/ip6_tunnel.ko
-  AUTOLOAD:=$(call AutoLoad,32,ip6_tunnel)
+  AUTOLOAD:=$(call AutoProbe,ip6_tunnel)
 endef
 
 define KernelPackage/ip6-tunnel/description
@@ -456,7 +456,7 @@ define KernelPackage/gre
   DEPENDS:=+PACKAGE_kmod-ipv6:kmod-ipv6 +kmod-iptunnel
   KCONFIG:=CONFIG_NET_IPGRE CONFIG_NET_IPGRE_DEMUX
   FILES:=$(LINUX_DIR)/net/ipv4/ip_gre.ko $(LINUX_DIR)/net/ipv4/gre.ko
-  AUTOLOAD:=$(call AutoLoad,39,gre ip_gre)
+  AUTOLOAD:=$(call AutoProbe,gre ip_gre)
 endef
 
 define KernelPackage/gre/description
@@ -487,7 +487,7 @@ define KernelPackage/tun
   TITLE:=Universal TUN/TAP driver
   KCONFIG:=CONFIG_TUN
   FILES:=$(LINUX_DIR)/drivers/net/tun.ko
-  AUTOLOAD:=$(call AutoLoad,30,tun)
+  AUTOLOAD:=$(call AutoProbe,tun)
 endef
 
 define KernelPackage/tun/description
@@ -698,7 +698,7 @@ define KernelPackage/sched-core
 	CONFIG_NET_EMATCH=y \
 	CONFIG_NET_EMATCH_U32
   FILES:=$(SCHED_FILES)
-  AUTOLOAD:=$(call AutoLoad,70, $(SCHED_MODULES_CORE))
+  AUTOLOAD:=$(call AutoProbe, $(SCHED_MODULES_CORE))
 endef
 
 define KernelPackage/sched-core/description
@@ -714,7 +714,7 @@ define KernelPackage/sched-connmark
   DEPENDS:=+kmod-sched-core +kmod-ipt-core +kmod-ipt-conntrack-extra
   KCONFIG:=CONFIG_NET_ACT_CONNMARK
   FILES:=$(LINUX_DIR)/net/sched/act_connmark.ko
-  AUTOLOAD:=$(call AutoLoad,71, act_connmark)
+  AUTOLOAD:=$(call AutoProbe, act_connmark)
 endef
 $(eval $(call KernelPackage,sched-connmark))
 
@@ -834,7 +834,7 @@ define KernelPackage/l2tp
 	CONFIG_L2TP_DEBUGFS=n
   FILES:=$(LINUX_DIR)/net/l2tp/l2tp_core.ko \
 	$(LINUX_DIR)/net/l2tp/l2tp_netlink.ko
-  AUTOLOAD:=$(call AutoLoad,32,l2tp_core l2tp_netlink)
+  AUTOLOAD:=$(call AutoProbe,l2tp_core l2tp_netlink)
 endef
 
 define KernelPackage/l2tp/description
--- ./package/kernel/linux/modules/netfilter.mk	2014-11-23 11:18:48.151374771 +0800
+++ ./package/kernel/linux/modules/netfilter.mk	2014-11-23 11:27:59.321287453 +0800
@@ -169,7 +169,7 @@ define KernelPackage/ipt-nat6
   TITLE:=IPv6 NAT targets
   KCONFIG:=$(KCONFIG_IPT_NAT6)
   FILES:=$(foreach mod,$(IPT_NAT6-m),$(LINUX_DIR)/net/$(mod).ko)
-  AUTOLOAD:=$(call AutoLoad,43,$(notdir $(IPT_NAT6-m)))
+  AUTOLOAD:=$(call AutoProbe,$(notdir $(IPT_NAT6-m)))
   $(call AddDepends/ipt,+kmod-ipt-conntrack)
   $(call AddDepends/ipt,+kmod-ipt-nat)
   $(call AddDepends/ipt,+kmod-ip6tables)
@@ -445,7 +445,7 @@ define KernelPackage/ip6tables
   DEPENDS:=+kmod-ipv6 +kmod-ipt-core +kmod-ipt-conntrack
   KCONFIG:=$(KCONFIG_IPT_IPV6)
   FILES:=$(foreach mod,$(IPT_IPV6-m),$(LINUX_DIR)/net/$(mod).ko)
-  AUTOLOAD:=$(call AutoLoad,42,$(notdir $(IPT_IPV6-m)))
+  AUTOLOAD:=$(call AutoProbe,$(notdir $(IPT_IPV6-m)))
 endef
 
 define KernelPackage/ip6tables/description
@@ -460,7 +460,7 @@ define KernelPackage/ip6tables-extra
   DEPENDS:=+kmod-ip6tables
   KCONFIG:=$(KCONFIG_IPT_IPV6_EXTRA)
   FILES:=$(foreach mod,$(IPT_IPV6_EXTRA-m),$(LINUX_DIR)/net/$(mod).ko)
-  AUTOLOAD:=$(call AutoLoad,43,$(notdir $(IPT_IPV6_EXTRA-m)))
+  AUTOLOAD:=$(call AutoProbe,$(notdir $(IPT_IPV6_EXTRA-m)))
 endef
 
 define KernelPackage/ip6tables-extra/description
