--- ./target/linux/generic/config-3.10	2014-08-31 13:14:21.000000000 +0800
+++ ./target/linux/generic/config-3.10	2014-08-31 13:41:39.788624286 +0800
@@ -1142,12 +1142,12 @@ CONFIG_HW_PERF_EVENTS=y
 # CONFIG_HW_RANDOM_VIA is not set
 # CONFIG_HYPERV is not set
 # CONFIG_HYSDN is not set
-CONFIG_HZ=100
-CONFIG_HZ_100=y
+CONFIG_HZ=250
+# CONFIG_HZ_100 is not set
 # CONFIG_HZ_1000 is not set
 # CONFIG_HZ_1024 is not set
 # CONFIG_HZ_128 is not set
-# CONFIG_HZ_250 is not set
+CONFIG_HZ_250=y
 # CONFIG_HZ_256 is not set
 # CONFIG_HZ_300 is not set
 # CONFIG_HZ_48 is not set
