--- ./package/libs/lzo/Makefile	2014-11-24 19:48:28.000000000 +0800
+++ ./package/libs/lzo/Makefile	2014-11-24 19:54:20.149753420 +0800
@@ -44,6 +44,10 @@ endef
 TARGET_CFLAGS += $(FPIC)
 MAKE_FLAGS += CFLAGS_O="$(TARGET_CFLAGS)"
 
+ifeq ($(CONFIG_i386)$(CONFIG_x86_64),)
+	TARGET_CFLAGS += -DLZO_CFG_NO_UNALIGNED=1
+endif
+
 define Build/InstallDev
 	$(INSTALL_DIR) $(1)/usr/include $(1)/usr/lib
 	$(CP) $(PKG_INSTALL_DIR)/usr/include/lzo $(1)/usr/include/
