--- ./package/libs/openssl/Makefile	2015-01-25 09:10:58.000000000 +0800
+++ ./package/libs/openssl/Makefile	2015-01-25 09:11:11.281966231 +0800
@@ -8,7 +8,7 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=openssl
-PKG_VERSION:=1.0.1k
+PKG_VERSION:=1.0.2
 PKG_RELEASE:=1
 PKG_USE_MIPS16:=0
 
@@ -18,7 +18,7 @@ PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).t
 PKG_SOURCE_URL:=http://www.openssl.org/source/ \
 	ftp://ftp.funet.fi/pub/crypt/mirrors/ftp.openssl.org/source \
 	ftp://ftp.sunet.se/pub/security/tools/net/openssl/source/
-PKG_MD5SUM:=d4f002bd22a56881340105028842ae1f
+PKG_MD5SUM:=38373013fc85c790aabf8837969c5eba
 
 PKG_LICENSE:=SSLEAY OPENSSL
 PKG_LICENSE_FILES:=LICENSE

--- ./package/libs/openssl/patches/200-parallel_build.patch	2015-01-24 13:53:10.000000000 +0800
+++ ./package/libs/openssl/patches/200-parallel_build.patch	2015-01-25 09:19:07.363461648 +0800
@@ -105,7 +105,7 @@
 +	+@target=all; $(RECURSIVE_MAKE)
  
  files:
- 	$(PERL) $(TOP)/util/files.pl Makefile >> $(TOP)/MINFO
+ 	$(PERL) $(TOP)/util/files.pl "CPUID_OBJ=$(CPUID_OBJ)" Makefile >> $(TOP)/MINFO
 -	@target=files; $(RECURSIVE_MAKE)
 +	+@target=files; $(RECURSIVE_MAKE)
  
