--- ./include/cmake.mk	2015-01-31 21:20:07.000000000 +0800
+++ ./include/cmake.mk	2015-02-09 20:24:15.393438808 +0800
@@ -39,7 +39,7 @@ define Build/Configure/Default
 			-DCMAKE_SYSTEM_NAME=Linux \
 			-DCMAKE_SYSTEM_VERSION=1 \
 			-DCMAKE_SYSTEM_PROCESSOR=$(ARCH) \
-			-DCMAKE_BUILD_TYPE=Release \
+			-DCMAKE_BUILD_TYPE=MinSizeRel \
 			-DCMAKE_C_FLAGS_RELEASE="-DNDEBUG" \
 			-DCMAKE_CXX_FLAGS_RELEASE="-DNDEBUG" \
 			-DCMAKE_C_COMPILER="$(CMAKE_C_COMPILER)" \
@@ -59,6 +59,8 @@ define Build/Configure/Default
 			-DCMAKE_INSTALL_PREFIX=/usr \
 			-DDL_LIBRARY=$(STAGING_DIR) \
 			-DCMAKE_PREFIX_PATH=$(STAGING_DIR) \
+			-DCMAKE_SKIP_RPATH=TRUE  \
+			-DCMAKE_ENABLE_NLS=0 \
 			$(CMAKE_OPTIONS) \
 		$(CMAKE_SOURCE_DIR) \
 	)
--- ./package/utils/util-linux/Makefile	2015-02-12 19:34:24.000000000 +0800
+++ ./package/utils/util-linux/Makefile	2015-02-12 19:34:37.514109416 +0800
@@ -31,6 +31,7 @@ define Package/util-linux/Default
 endef
 
 CONFIGURE_ARGS += \
+	--disable-rpath	\
 	--enable-new-mount	\
 	--with-ncurses		\
 	--disable-tls		\
--- ./rules.mk	2015-01-31 21:20:07.000000000 +0800
+++ ./rules.mk	2015-02-12 19:39:18.375441515 +0800
@@ -254,6 +254,7 @@ else
     NM="$(TARGET_CROSS)nm" \
     STRIP="$(STRIP)" \
     STRIP_KMOD="$(SCRIPT_DIR)/strip-kmod.sh" \
+    PATCHELF="$(STAGING_DIR_HOST)/bin/patchelf" \
     $(SCRIPT_DIR)/rstrip.sh
 endif
 
--- ./scripts/rstrip.sh	2015-01-31 21:20:07.000000000 +0800
+++ ./scripts/rstrip.sh	2015-02-12 19:41:00.675387546 +0800
@@ -25,11 +25,21 @@ find $TARGETS -type f -a -exec file {} \
 (
   IFS=":"
   while read F S; do
-    echo "$SELF: $F:$S"
+    echo "$SELF: $F: $S"
 	[ "${S}" = "relocatable" ] && {
 		eval "$STRIP_KMOD $F"
 	} || {
 		b=$(stat -c '%a' $F)
+		[ -z "$PATCHELF" ] || [ -z "$TOPDIR" ] || { 
+	                        old_rpath="$($PATCHELF --print-rpath $F)"; new_rpath="" 
+	                        for path in $old_rpath; do 
+	                                case "$path" in 
+	                                        /lib/[^/]*|/usr/lib/[^/]*|\$ORIGIN/*) new_rpath="${new_rpath:+$new_rpath:}$path" ;; 
+	                                        *) echo "$SELF: $F: removing rpath $path" ;; 
+	                                esac 
+	                        done 
+	                        [ "$new_rpath" = "$old_rpath" ] || $PATCHELF --set-rpath "$new_rpath" $F 
+	                } 
 		eval "$STRIP $F"
 		a=$(stat -c '%a' $F)
 		[ "$a" = "$b" ] || chmod $b $F
